src/constants/action-types.js

```diff
@@ -27,6 +27,12 @@ export const SEARCH_AUTHORS_REQUEST = 'SEARCH_AUTHORS_REQUEST'
 export const SEARCH_AUTHORS_SUCCESS = 'SEARCH_AUTHORS_SUCCESS'
 export const SEARCH_AUTHORS_FAILURE = 'SEARCH_AUTHORS_FAILURE'
 
+export const LIST_ALL_AUTHORS_REQUEST = 'LIST_ALL_AUTHORS_REQUEST'
+export const LIST_ALL_AUTHORS_SUCCESS = 'LIST_ALL_AUTHORS_SUCCESS'
+export const LIST_ALL_AUTHORS_FAILURE = 'LIST_ALL_AUTHORS_FAILURE'
+
+export const SET_AUTHORS_LIST_TYPE = 'SET_AUTHORS_LIST_TYPE'
```

Comment:
listing all authors or only listing searched authors could be maintained in the `state` of authors container. 
Hence, we don't need `LIST_ALL_AUTHORS_SUCCESS ` and `SET_AUTHORS_LIST_TYPE ` action type.